annotations/tasks.py

```diff
@@ -278,3 +293,69 @@ def handle_file_upload(request, form):
     if file_content != None:
         tokenized_content = tokenize(file_content)
         return save_text_instance(tokenized_content, text_title, date_created, is_public, user, uri)
+
+
+
+
+
+
+@shared_task
+def process_import_task(user, project_id ,file, part_of_id, action, name):
+    with open(file, 'r') as f:
+        csv_file = csv.reader(f)
+        # there isn't a good way to reset the iterator so we tee them up
+        reader1, reader2 = tee(csv_file)
```

Comment:
you should be able to set the filereader to its beginning to reset the iterator: f.seek(0) and then recreate the reader. this works probably too though. just name them better.